packages/cosmic-swingset/test/test-make.js

```diff
@@ -43,3 +68,92 @@ test('make and exec', async t => {
     }),
   );
 });
+
+const query = async args => {
+  const parts = [];
+  const run = new Promise((resolve, reject) => {
+    const cmd = spawn(
+      'bin/ag-chain-cosmos',
+      ['--chain-id=agoric', 'query', ...args],
+      { cwd: `${dirname}/..` },
+    );
+    cmd.stdout.on('data', chunk => parts.push(chunk));
+    cmd.on('exit', code => {
+      if (code !== 0) {
+        reject(Error(`exit ${code} from: query ${args}`));
+        return;
+      }
+      const txt = parts.join('');
+      if (txt.trim() === '') {
+        reject(Error(`empty output from: query ${args}`));
+        return;
+      }
+      const info = JSON.parse(txt);
+      resolve(info);
+    });
+  });
+  return run;
+};
+
+const waitForBlock = async height => {
+  const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+  for (;;) {
+    try {
+      const info = await query(['block']);
+      if (Number(info.block.header.height) >= height) {
+        return;
+      }
+    } catch (reason) {
+      console.warn(reason);
+      await delay(5_000);
+    }
+  }
+};
+
+test.serial('PSM bootstrap and provision', async t => {
+  // XXX coordinate with vbank/provision from golang
+  const pool = 'agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346';
+
+  const addrLine = await fs.promises.readFile(
+    './t1/8000/ag-cosmos-helper-address',
+    'utf-8',
+  );
+  const addr = addrLine.trim();
```

Comment:
since you're immediately awaiting, might as well read sync. and no need to bind addrLine.
```suggestion
  const addr = fs.readFileSync(
    './t1/8000/ag-cosmos-helper-address',
    'utf-8',
  ).trim();
```

Reply:
synchronous I/O messes with turn isolation. It should only be used in low-level stuff that's building a platform for normal code.

> The net effect is that a turn is E’s unit of operation. We can faithfully account for the visible effects of concurrency without any interleaving of the steps within a turn. Any actual multi-vat computation is equivalent to some fully ordered
interleaving of turns.
> The actual E system does provide synchronous file I/O operations. When these files are local, prompt, and private to the vat accessing them, this does not violate turn isolation, but since files may be remote, non-prompt, or shared, the availability of these synchronous I/O operations does violate the E model -- [Concurrency Among Strangers](http://papers.agoric.com/assets/pdf/papers/concurrency-among-strangers.pdf)

Reply:
Thanks for the lesson. I fail to see how it applies here as this is a test, running in a single VM, serially.

Not opposed though